### PR TITLE
feat(contract-mixins): Add Pausable

### PR DESCRIPTION
**Pull request by davemneo**
### Requirements

Create a mix-in for Pausable contract.

### CODE REVIEW

**Comment by davemneo** [comment]
Currently unable to build contract in test, I've added some console logs to the test.

Possibly related to nested class inheritance, see: https://github.com/neo-one-suite/neo-one/pull/1103/files#diff-db0e6bd7ba5752ace5c640c8ab17b191R6
(or a bad import) 
Files to pay attention to: 
- packages/neo-one-smart-contract-lib/src/__data__/contracts/lifecycle/TestPausable.ts
- packages/neo-one-smart-contract-lib/src/__tests__/lifecycle/TestPausable.test.ts
- packages/neo-one-smart-contract-lib/src/access/index.ts
- packages/neo-one-smart-contract-lib/src/index.ts
- packages/neo-one-smart-contract-lib/src/lifecycle/Pausable.ts

File: gulpfile.js [hunk 1]
```diff
@@ -408,7 +408,7 @@ const copyTypescript = ((cache) =>
       .src(globs.typescript)
       .pipe(
         gulpFilter([
-          'packages/neo-one-smart-contract-lib/src/*.ts',
+          'packages/neo-one-smart-contract-lib/src/**/*.ts',
```
Comment (danwbyrne) [inline gulpfile.js]: You should maybe split this change into a separate PR as I think it will affect more than just this one. @davemneo

File: packages/neo-one-smart-contract-lib/src/__data__/contracts/lifecycle/TestPausable.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { Address, SmartContract, constant } from '@neo-one/smart-contract';
+/* tslint:disable-next-line: no-implicit-dependencies */
```
Comment (danwbyrne) [inline packages/neo-one-smart-contract-lib/src/__data__/contracts/lifecycle/TestPausable.ts]: Is there a way we can add a *solid* TSLint exclude for these files? Seeing this in every `__data__` file is a bit unintuitive. Not sure if we have that control for single rule ignoring across a glob.

File: packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: transfer consume 1`] = `"0"`;
 
-exports[`RedToken properties + issue + balanceOf + totalSupply + transfer: transfer cost 1`] = `"3.389"`;
```
Comment (danwbyrne) [inline packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap]: Curious: does your contract affect the RedToken test? Odd that it makes it cost more gas, no?

File: tsconfig.json [hunk 4]
```diff
@@ -37,16 +37,22 @@
   "exclude": [
     "dist",
     "packages/neo-one-server-plugin-wallet/src/contracts/*.ts",
+    "packages/neo-one-smart-contract-lib/src/access/roles/*.ts",
```
Comment (danwbyrne) [inline tsconfig.json]: I think you should just use
```
"packages/neo-one-smart-contract-lib/src/access/**/*.ts,
```
instead of
```
"packages/neo-one-smart-contract-lib/src/access/*.ts",
"packages/neo-one-smart-contract-lib/src/access/roles/*.ts"
```

could probably do the same for lifecycles.

File: packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap (hunk 3 above)
Comment (davemneo) [reply packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap]: Shouldn't be.  I'll look into this.  Did any Opcode changes go in?

File: gulpfile.js (hunk 1 above)
Comment (davemneo) [reply gulpfile.js]: I applied it to affected PR's so I don't have to wait for the PR and CI process.

File: packages/neo-one-smart-contract-lib/src/__data__/contracts/lifecycle/TestPausable.ts (hunk 2 above)
Comment (davemneo) [reply packages/neo-one-smart-contract-lib/src/__data__/contracts/lifecycle/TestPausable.ts]: @danwbyrne totally agree, but it looks like they don't want to include a Conditional configuration [https://github.com/palantir/tslint/issues/3447]

File: tsconfig.json (hunk 4 above)
Comment (davemneo) [reply tsconfig.json]: done

File: packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap (hunk 3 above)
Comment (davemneo) [reply packages/neo-one-smart-contract-lib/src/__tests__/__snapshots__/RedToken.test.ts.snap]: This happens because the compiler is grabbing everything that's being exported by `@neo-one/smart-contracts-lib` -- Alex indicated there's a plan to clean this up.

**Comment by davemneo** [comment]
New structure, new PRs to inherit from.
